internal/server/iifo.go

```diff
@@ -15,3 +22,58 @@ func (s *SOMASServer) runIIFOEndOfTurn() error {
 	// TODO:- IIFO team
 	return nil
 }
+
+func (s *SOMASServer) runPredictionSession() error {
+	s.logf("start runPredictionSession")
+	islandPredictionDict, err := s.getPredictions()
+	if err != nil {
+		return err
+	}
+
+	err = s.distributePredictions(islandPredictionDict)
+	if err != nil {
+		return err
+	}
+
+	defer s.logf("finish runPredictionSession")
+	return nil
+}
+
+func (s *SOMASServer) getPredictions() (shared.PredictionInfoDict, error) {
+	islandPredictionsDict := shared.PredictionInfoDict{}
+	var err error
+	for id, client := range s.clientMap {
+		islandPredictionsDict[id], err = client.MakePrediction()
+		if err != nil {
+			return islandPredictionsDict, err

```

Comment:
Let's wrap the err
```suggestion
			return islandPredictionsDict, errors.Errorf("Failed to get prediction from %v: %v", id, err)
```

Reply:
Do we also run this on dead clients?

See `GameStateUpdate` on how dead clients are not reached out to.

Reply:
Would it be easier to make another map with only the alive clients in, as most actions will only be done on these clients anyway it seems abit unneccessary to always have to check if a client is dead

Reply:
Yep, go ahead and make a server function for `getNonDeadClients` 👍